strategy/json/JSONAPIStrategy.cfc

```diff
@@ -0,0 +1,152 @@
+/**
+* Default Document Strategy for DocBox
+* <br>
+* <small><em>Copyright 2015 Ortus Solutions, Corp <a href="www.ortussolutions.com">www.ortussolutions.com</a></em></small>
+*/
+component extends="docbox.strategy.AbstractTemplateStrategy" accessors="true"{
+
+	/**
+	* The output directory
+	*/
+	property name="outputDir" type="string";
+
+	/**
+	* The project title to use
+	*/
+	property name="projectTitle" default="Untitled" type="string";
+
+	// Static variables.
+	variables.static.TEMPLATE_PATH	= "/docbox/strategy/json/resources/templates";
+	variables.static.ASSETS_PATH 	= "/docbox/strategy/json/resources/static";
+
+	/**
+	* Constructor
+	* @outputDir The output directory
+	* @projectTitle The title used in the HTML output
+	*/
+	component function init( required outputDir, string projectTitle="Untitled" ){
+		super.init();
+
+		variables.outputDir 	= arguments.outputDir;
+		variables.projectTitle 	= arguments.projectTitle;
+
+		return this;
+	}
+
+	/**
+	* Run this strategy
+	* @qMetaData The metadata
+	*/
+	component function run( required query qMetadata ){
+		ensureDirectory( getOutputDir() );
+		ensureDirectory( getOutputDir() & "/classes" );
+
+		//write the index template
+		var args = {
+			path 		 = getOutputDir() & "/index.json", 
+			template 	 = "#variables.static.TEMPLATE_PATH#/index.cfm", 
+			projectTitle = getProjectTitle()
+		};
+
+		// consider chainable methods, since that's how HTMLAPIStrategy works.
+		var classData = normalize( queryToArray( arguments.qMetadata ) );
+
+		serializeToFile(
+			getOutputDir() & "/index.json",
+			generateClassIndex( classData )
+		);
+
+		classData.each( ( class ) => {
+			serializeToFile(
+				getOutputDir() & "/classes/#class.name#.json",
+				class
+			);
+		});
+
+		return this;
+	}
+
+	/**
+	 * Marshall component metadata into a class index file.
+	 * This will be spit into an `index.json` file at the root of the output directory.
+	 *
+	 * @classData 
+	 */
+	package struct function generateClassIndex( required array classData ){
+		var classMap = arguments.classData.map( ( class ) => {
+			return {
+				"name" : class.name,
+				"path" : "#getOutputDir()#/classes/#class.name#.json"
+			}
+		});
+		return {
+			"packages" : classMap
+		}
+	}
+
+	package array function normalize( required array classData ){
+		var componentMeta = [ "hint", "extends", "functions", "fullname", "properties", "remoteAddress", "type", "hashCode", "name", "path" ];
+		var functionMeta = [ "name", "hint", "description", "access", "closure", "owner", "position", "returnFormat", "returnType", "modifier", "parameters" ];
+		return arguments.classData.map( function( row ) {
+			/**
+			 * Marshall component metadata into attributes array.
+			 */
+			var metaAttributes = row.metadata.filter( ( key ) => {
```

Comment:
I would stay away from `attributes`, like we discussed in our meeting.

Basically take the MD as is just make sure you add defaults for sections that we always use, so the output json can contain defaults if not defined in the MD.  This goes the same for functinos.

Reply:
:+1: I took care of this, just forgot to push my commit up to the PR branch. :roll_eyes: